hplsql/src/main/java/org/apache/hive/hplsql/Stmt.java

```diff
@@ -383,8 +383,11 @@ public Integer drop(HplsqlParser.Drop_stmtContext ctx) {
         sql += "IF EXISTS ";
       }
       sql += evalPop(ctx.table_name()).toString();
-    }
-    else if (ctx.T_DATABASE() != null || ctx.T_SCHEMA() != null) {
+    } else if (ctx.T_PACKAGE() != null) {
+      exec.dropPackage(ctx.ident().getText().toUpperCase(), ctx.T_EXISTS() != null);
```

Comment:
why don't we pass the context to these two functions too? That way we can get line info in trace call.

Reply:
I added the context.